llvm/lib/Object/OffloadBinary.cpp

```diff
@@ -167,6 +216,100 @@ Error extractFromArchive(const Archive &Library,
 
 } // namespace
 
+Error OffloadBundleFatBin::ReadEntries(StringRef Buffer,
+                                       uint64_t SectionOffset) {
+  uint64_t BundleNumber = 0;
+  uint64_t NumOfEntries = 0;
+
+  // get Reader
+  BinaryStreamReader Reader(Buffer, llvm::endianness::little);
+
+  // Read the Magic String first.
+  StringRef Magic;
+  if (auto EC = Reader.readFixedString(Magic, 24)) {
+    return errorCodeToError(object_error::parse_failed);
+  }
```

Comment:
```suggestion
  if (auto EC = Reader.readFixedString(Magic, 24))
    return errorCodeToError(object_error::parse_failed);
```
And everywhere else.

Reply:
👍